lib/global/base.rb

```diff
@@ -59,61 +77,21 @@ def generate_js(options = {})
 
     protected
 
-    def load_configuration(dir, env)
-      config = load_from_file(dir, env)
-      config.deep_merge!(load_from_directory(dir, env))
-      Configuration.new(config)
-    end
-
-    def load_from_file(dir, env)
-      config = {}
-
-      if File.exist?(file = "#{dir}#{YAML_EXT}")
-        configurations = load_yml_file(file)
-        config = get_config_by_key(configurations, 'default')
-        config.deep_merge!(get_config_by_key(configurations, env))
-        if File.exist?(env_file = "#{dir}#{FILE_ENV_SPLIT}#{env}#{YAML_EXT}")
-          config.deep_merge!(load_yml_file(env_file) || {})
-        end
-      end
-
-      config
-    end
-
-    def get_config_by_key(config, key)
-      config[key.to_sym] || config[key.to_s] || {}
-    end
-
-    def load_yml_file(file)
-      YAML.safe_load(
-        ERB.new(IO.read(file)).result,
-        [Date, Time, DateTime, Symbol].concat(yaml_whitelist_classes),
-        [], true
-      )
-    end
-
-    def load_from_directory(dir, env)
-      config = {}
-
-      if File.directory?(dir)
-        Dir["#{dir}/*"].each do |entry|
-          namespace = File.basename(entry, YAML_EXT)
-          next if namespace.include? FILE_ENV_SPLIT # skip files with dot(s) in name
-
-          file_with_path = File.join(File.dirname(entry), File.basename(entry, YAML_EXT))
-          config.deep_merge!(namespace => load_configuration(file_with_path, env))
-        end
-      end
-
-      config
-    end
-
     def respond_to_missing?(method, include_private = false)
       configuration.key?(method) || super
     end
 
     def method_missing(method, *args, &block)
-      configuration.key?(method) ? configuration[method] : super
+      configuration.send(method)
```

Comment:
Not clear why this changed. `send` can create security problem

Reply:
it's because I don't pre-wrap the nested hashes in Configuration, so if I don't use the `method_missing` on Configuration, I get back unwrapped hashes, and not nested Configuration objects.

Is bringing back the check for `configuration.key?(method)` enough to make it secure?

Reply:
Possible, that `public_send` will be safe, but this need to be checked.